modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/sidebar/Sidebar.es.js

```diff
@@ -15,98 +15,98 @@
 
-	if (!isOpen) {
-		return null;
-	}
-
 	const {icon, label} = field;
 
 	return (
 		<>
-			<div className="sidebar-backdrop"></div>
-			<div className="open sidebar-container sidebar-reports">
-				<div className="sidebar sidebar-light">
-					<nav className="component-tbar tbar">
-						<div className="container-fluid">
-							<ul className="tbar-nav">
-								<li className="tbar-item">
-									<div className="icon">
-										<ClayIcon symbol={icon} />
-									</div>
-								</li>
-
-								<li className="tbar-item tbar-item-expand">
-									<div className="tbar-section">
-										<div className="field-info">
-											<p className="title">{label}</p>
-
-											<p className="description">
-												{totalEntries}{' '}
-												{Liferay.Language.get(
-													'entries'
-												).toLowerCase()}
-											</p>
-										</div>
-									</div>
-								</li>
-
-								<li className="tbar-item">
-									<ClayButton
-										className="close-sidebar"
-										displayType="secondary"
-										monospaced
-										onClick={() => toggleSidebar()}
-									>
-										<ClayIcon
-											className="close-button"
-											symbol={'times-small'}
-										/>
-									</ClayButton>
-								</li>
-							</ul>
-						</div>
-					</nav>
-
-					<div className="sidebar-body">
-						{isLoading && (
-							<div className="align-items-center d-flex loading-wrapper">
-								<ClayLoadingIndicator />
+			<nav className="component-tbar tbar">
+				<div className="container-fluid">
+					<ul className="tbar-nav">
+						<li className="tbar-item">
+							<div className="icon">
+								<ClayIcon symbol={icon} />
 							</div>
-						)}
+						</li>
+
+						<li className="tbar-item tbar-item-expand">
+							<div className="tbar-section">
+								<div className="field-info">
+									<p className="title">{label}</p>
 
-						<List data={data} type={type}></List>
+									<p className="description">
+										{totalEntries}{' '}
+										{Liferay.Language.get(
+											'entries'
+										).toLowerCase()}
+									</p>
+								</div>
+							</div>
+						</li>
+
+						<li className="tbar-item">
+							<ClayButton
+								className="close-sidebar"
+								displayType="secondary"
+								monospaced
+								onClick={() => toggleSidebar()}
+							>
+								<ClayIcon
+									className="close-button"
+									symbol={'times-small'}
+								/>
+							</ClayButton>
+						</li>
+					</ul>
+				</div>
+			</nav>
+
+			<div className="sidebar-body">
+				{isLoading && (
+					<div className="align-items-center d-flex loading-wrapper">
+						<ClayLoadingIndicator />
 					</div>
+				)}
+
+				<List data={data} type={type}></List>
+			</div>
+		</>
+	);
+};
+
+export default () => {
+	const {field, isOpen} = useContext(SidebarContext);
+
+	const ref = useRef();
+
+	return (
+		<>
+			{isOpen && <div className="sidebar-backdrop"></div>}
```

Comment:
@marcosapmf, can you please clarify the purpose of add this `div` here?

Reply:
Hi, @natocesarrego,

This `div` already existed before: [Sidebar.es.js#L59](https://github.com/liferay/liferay-portal/blob/3994ec2ec702175c32939899392b6159acc423f3/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/sidebar/Sidebar.es.js#L59)

I believe that this class (sidebar-backdrop) has the purpose of changing the opacity of the background when the sidebar is open.

Is this correct @marcosapmf ?

Reply:
I see, @carolmariaabb. My question is more related to the purpose of add it in this check: `{isOpen && <div ... />}`. Sorry if it was not clear. 😞

Reply:
I believe it's to keep the same logic from here [Sidebar.es.js#L51-L53](https://github.com/liferay/liferay-portal/blob/3994ec2ec702175c32939899392b6159acc423f3/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/sidebar/Sidebar.es.js#L51-L53).

But Marcos can answer that better 👀

Reply:
@natocesarrego, we add this div only when the sidebar is open, it changes the opacity of the background, as @carolmariaabb well said. Therefore, once the sidebar is closed the div should not be rendered because if so, we could not click anywhere on the screen :)

Reply:
Thanks for your detailed explanation, guys! 😄